### PR TITLE
CID-2030: Add repository scheduler

### PR DESCRIPTION
**Pull request by henriqamaral**

### CODE REVIEW

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/scheduler/RepositoryScheduler.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.stereotype.Component
+
+@Component
+class RepositoryScheduler(
+    private val initialStateService: InitialStateService,
+) {
+
+    private val logger = LoggerFactory.getLogger(RepositoryScheduler::class.java)
+
+    @Scheduled(cron = "\${leanix.vsm.schedule:0 0 4 * * *}")
```
Comment (ahmed-ali-55) [inline src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/scheduler/RepositoryScheduler.kt:16]: Can we have an env variable to override it an use `0 0 4 * * *` as default?

**Review by ahmed-ali-55 — APPROVED** [review]
Just that one change

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/scheduler/RepositoryScheduler.kt (hunk 1 above)
Comment (henriqamaral) [reply src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/scheduler/RepositoryScheduler.kt:16]: It's a good point, let's discuss it on the daily

Comment (ahmed-ali-55) [reply src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/scheduler/RepositoryScheduler.kt:16]: looks like we won't be doing this.
marking it as resolved

**Pull request merged**